src/browser-connection/local.js

```diff
@@ -28,11 +29,29 @@ export default class LocalBrowserConnection extends BrowserConnection {
         });
     }
 
+    _waitForClose () {
+        return new Promise(resolve => {
+            if (this.disconnected) {
+                resolve();
+                return;
+            }
+
+            var waitingTimeout = setTimeout(() => {
+                resolve();
+            }, this.WAITING_FOR_CLOSE_TIMEOUT);
+
+            this.once('closed', () => {
+                clearTimeout(waitingTimeout);
+                resolve();
+            });
+        });
+    }
 
-    close () {
+    async close () {
```

Comment:
Then we can encapsulate closing logic for the bunch of `LocalConnections` in `freeBrowserConnections` methods (like we have right now in the `runner`). But seems like something very similar currently used in `bootsrapper` as well. So, I think we should move it to the separate reusable module. I'm starting to think, that we should implement something like `BrowserConnectionSet`, which will server all this need during browser connections life span. E.g.: bootsrapper adds all newly established and existend BC's to the set. Set has `error` event, so both bootstrapper and runner could subscribe to it and fail appropriately, without a need to subscribe/unsubscribe from each BC. Set also provides async method to close local connections. Moreover, we can make task accept Set as browser list, instead of array. I'm not sure about `Set` name, need to discuss with @VasilyStrelyaev